doc/dev/crimson/pipelines.rst

```diff
@@ -0,0 +1,258 @@
+=========
+Pipelines
+=========
+
+RADOS Protocol Requirements
+===========================
+
+Requests
+--------
+
+OSD serves different kinds of requests. Some of them are targeting the local state of the connected
+OSD instance. For instance, the ``ceph config get`` command is directed straight to the local config
+subsystem, and is fulfilled right away. But others inquiry or mutate the shared state of the cluster.
+Following are two categories of these requests:
+
+- I/O requests from clients.
+- PG peering requests from peer OSDs.
```

Comment:
PG peering requests are tough to discuss in the general case here.  We make use of OSD<->OSD message ordering to reduce the amount of complexity in the peering protocol and other messages, but we could still implement the peering/recovery/etc protocols without those ordering properties -- it would just be much more complicated.

Reply:
removed. let's just focus on i/o requests in this document.